rest-api/src/main/scala/za/co/absa/enceladus/rest_api/services/v3/VersionedModelServiceV3.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.rest_api.services.v3
+
+import org.mongodb.scala.result.UpdateResult
+import org.slf4j.LoggerFactory
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.core.userdetails.UserDetails
+import za.co.absa.enceladus.model.{UsedIn, Validation}
+import za.co.absa.enceladus.model.menas.audit._
+import za.co.absa.enceladus.model.versionedModel.{VersionedModel, VersionedSummary}
+import za.co.absa.enceladus.rest_api.exceptions._
+import za.co.absa.enceladus.rest_api.services.VersionedModelService
+
+import scala.concurrent.Future
+
+// scalastyle:off number.of.methods
+trait VersionedModelServiceV3[C <: VersionedModel with Product with Auditable[C]] extends VersionedModelService[C] {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  override private[services] val logger = LoggerFactory.getLogger(this.getClass)
+
+  def getLatestVersionSummary(name: String): Future[Option[VersionedSummary]] = {
+    mongoRepository.getLatestVersionSummary(name)
+  }
+
+  // v3 has internal validation on importItem (because it is based on update), v2 only had it on import
+  override def importSingleItem(item: C, username: String, metadata: Map[String, String]): Future[Option[(C, Validation)]] = {
```

Comment:
Why is the metadata passed here?

Reply:
Thanks for noticing and making me realizing the problem! 🙇‍♂️ 
The metadata is passed there, because it should have been checked, too, but this implementation failed to do so. 

Should be fixed now, I have updated the code and added test cases for both V2 and V3 import API to hold this together.